modules/core/src/persistence.cpp

```diff
@@ -693,8 +692,8 @@ class FileStorage::Impl : public FileStorage_API
         }
         else
         {
-            const size_t buf_size0 = 1 << 20;
-            size_t buf_size = buf_size0;
+            const size_t buf_size0 = 40;
+            buffer.resize(buf_size0);
```

Comment:
Do we left this "as is" for merge?

Reply:
@alalek, I suggest to merge it as-is. It will help to catch issues (if any) with this patch sooner rather than later.